functional_test.go

```diff
@@ -859,6 +864,62 @@ func TestGlobalRateLimits(t *testing.T) {
 	})
 }
 
+func TestGlobalRateLimitsWithLoadBalancing(t *testing.T) {
+	owner := cluster.PeerAt(2).GRPCAddress
+	peer := cluster.PeerAt(0).GRPCAddress
+	assert.NotEqual(t, owner, peer)
+
+	dialOpts := []grpc.DialOption{
+		grpc.WithResolvers(newStaticBuilder()),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"round_robin":{}}]}`),
+	}
+
+	address := fmt.Sprintf("static:///%s,%s", owner, peer)
+	conn, err := grpc.DialContext(context.Background(), address, dialOpts...)
+	require.NoError(t, err)
+
+	client := guber.NewV1Client(conn)
+
+	sendHit := func(status guber.Status, assertion func(resp *guber.RateLimitResp), i int) string {
+		ctx, cancel := context.WithTimeout(context.Background(), clock.Hour*5)
+		defer cancel()
+		resp, err := client.GetRateLimits(ctx, &guber.GetRateLimitsReq{
+			Requests: []*guber.RateLimitReq{
+				{
+					Name:      "test_global",
+					UniqueKey: "account:12345",
+					Algorithm: guber.Algorithm_LEAKY_BUCKET,
+					Behavior:  guber.Behavior_GLOBAL,
+					Duration:  guber.Minute * 5,
+					Hits:      1,
+					Limit:     2,
+				},
+			},
+		})
+		require.NoError(t, err, i)
+		gotResp := resp.Responses[0]
+		assert.Equal(t, "", gotResp.GetError(), i)
+		assert.Equal(t, status, gotResp.GetStatus(), i)
+
+		if assertion != nil {
+			assertion(gotResp)
+		}
+
+		return gotResp.GetMetadata()["owner"]
+	}
+
+	// Send two hits that should be processed by the owner and the peer and deplete the limit
+	sendHit(guber.Status_UNDER_LIMIT, nil, 1)
+	sendHit(guber.Status_UNDER_LIMIT, nil, 2)
+	// sleep to ensure the async forward has occurred and state should be shared
+	time.Sleep(time.Second * 5)
```

Comment:
I think (not sure) you can control this via setting `globalBatchTimeout = 1` and `globalSyncWait = 1ms`, so you don't need to wait so long here

Reply:
Or, use `github.com/mailgun/holster/v4/clock` to freeze or advance system time during the test.